src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,485 @@
+-- @
+isEmpty :: SymmetricRelation a -> Bool
+isEmpty = R.isEmpty . fromSymmetric
+
+-- | Check if a graph contains a given vertex.
+-- Complexity: /O(log(n))/ time.
+--
+-- @
+-- hasVertex x 'empty'            == False
+-- hasVertex x ('vertex' x)       == True
+-- hasVertex 1 ('vertex' 2)       == False
+-- hasVertex x . 'removeVertex' x == 'const' False
+-- @
+hasVertex :: Ord a => a -> SymmetricRelation a -> Bool
+hasVertex x = R.hasVertex x . fromSymmetric
+
+-- | Check if a graph contains a given edge.
+-- Complexity: /O(log(n))/ time.
+--
+-- @
+-- hasEdge x y 'empty'            == False
+-- hasEdge x y ('vertex' z)       == False
+-- hasEdge x y ('edge' x y)       == True
+-- hasEdge y x ('edge' x y)       == True
+-- hasEdge x y . 'removeEdge' x y == 'const' False
+-- hasEdge x y                    == 'elem' (x,y) . 'edgeList'
+-- @
+hasEdge :: Ord a => a -> a -> SymmetricRelation a -> Bool
+hasEdge x y = R.hasEdge x y . fromSymmetric
+
+-- | The number of vertices in a graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- vertexCount 'empty'             ==  0
+-- vertexCount ('vertex' x)        ==  1
+-- vertexCount                     ==  'length' . 'vertexList'
+-- vertexCount x \< vertexCount y  ==> x \< y
+-- @
+vertexCount :: SymmetricRelation a -> Int
+vertexCount = R.vertexCount . fromSymmetric
+
+-- | The number of edges in a graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- edgeCount 'empty'      == 0
+-- edgeCount ('vertex' x) == 0
+-- edgeCount ('edge' x y) == 1
+-- edgeCount              == 'length' . 'edgeList'
+-- @
+edgeCount :: Ord a => SymmetricRelation a -> Int
+edgeCount = length . edgeList
+
+-- | The sorted list of vertices of a given graph.
+-- Complexity: /O(n)/ time and memory.
+--
+-- @
+-- vertexList 'empty'      == []
+-- vertexList ('vertex' x) == [x]
+-- vertexList . 'vertices' == 'Data.List.nub' . 'Data.List.sort'
+-- @
+vertexList :: SymmetricRelation a -> [a]
+vertexList = R.vertexList . fromSymmetric
+
+-- | The sorted list of edges of a graph.
+-- Complexity: /O(n + m)/ time and /O(m)/ memory.
+--
+-- @
+-- edgeList 'empty'          == []
+-- edgeList ('vertex' x)     == []
+-- edgeList (edge x y)       == [(min x y, max y x)]
+-- edgeList ('star' 2 [3,1]) == [(1,2), (2,3)]
+-- @
+edgeList :: Ord a => SymmetricRelation a -> [(a, a)]
+edgeList = deduplicate . R.edgeList . fromSymmetric
+
+-- | The set of vertices of a given graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- vertexSet 'empty'      == Set.'Set.empty'
+-- vertexSet . 'vertex'   == Set.'Set.singleton'
+-- vertexSet . 'vertices' == Set.'Set.fromList'
+-- @
+vertexSet :: SymmetricRelation a -> Set.Set a
+vertexSet = R.vertexSet . fromSymmetric
+
+-- | The set of edges of a given graph.
+-- Complexity: /O(m)/ time.
+--
+-- @
+-- edgeSet 'empty'      == Set.'Set.empty'
+-- edgeSet ('vertex' x) == Set.'Set.empty'
+-- (edgeSet ('edge' x y) == Set.'Set.singleton' (x,y)) || (edgeSet ('edge' x y) == Set.'Set.singleton' (y,x))
```

Comment:
Let's simplify to `edgeSet ('edge' x y) == Set.'Set.singleton' (min x y, max x y)`, as above.